src/dtu_mlops_project/frontend.py

```diff
@@ -0,0 +1,118 @@
+import os
+
+import mimetypes
+import pandas as pd
+from PIL import Image
+import requests
+import streamlit as st
+from google.cloud import run_v2
+
+
+@st.cache_resource
+def get_backend_url():
+    """Get the URL of the backend service."""
+    parent = "projects/dtu-mlops-447711/locations/europe-west4"
+    client = run_v2.ServicesClient()
+    services = client.list_services(parent=parent)
+    for service in services:
+        if service.name.split("/")[-1] == "production-model":
+            return service.uri
+    return os.environ.get("BACKEND", None)
+
+
+def about_model(backend):
+    """Send a request to the backend to get information about the model."""
+    about_url = f"{backend}/about/"
+    response = requests.get(about_url)
+    if response.status_code == 200:
+        return response.json()
+    return None
+
+
+def classify_image(image, mime_type, backend):
+    """Send the image to the backend for classification."""
+    predict_url = f"{backend}/api/predict/"
+    # files = {"image": ("uploaded_image", image, mime_type)}
+    files = {"image_file": open(image.name, "rb")}
+    response = requests.post(predict_url, files=files, timeout=10)
+    if response.status_code == 200:
+        return response.json()
+    return None
+
+
+def main() -> None:
+    """Main function of the Streamlit frontend."""
+    backend = get_backend_url()
+    if backend is None:
```

Comment:
If I'm sneaky and I set the backend var to `""` (empty string), this validation logic won't catch it.

Maybe you'll want to use `if not backend:` instead?